app/lti.go

```diff
@@ -20,28 +20,60 @@ func (a *App) GetLMSToUse(consumerKey string) model.LMS {
 	return nil
 }
 
-func (a *App) OnboardLMSUser(userId string, lms model.LMS, launchData map[string]string) *model.AppError {
-	teamSlug := lms.GetTeam(launchData)
-	team, err := a.GetTeamByName(teamSlug)
-	if err != nil {
-		mlog.Error(fmt.Sprintf("Team to be used: %s could not be found: %s", teamSlug, err.Error()))
-		return model.NewAppError("OnboardLMSUser", "app.onboard_lms_user.team_not_found.app_error", nil, "", http.StatusInternalServerError)
+func (a *App) OnboardLTIUser(userId string, lms model.LMS, launchData map[string]string, joinChannels bool) *model.AppError {
+	teamName := lms.GetTeam(launchData)
+	if err := a.addTeamMemberIfRequired(userId, teamName); err != nil {
+		return err
 	}
 
-	if _, err := a.GetTeamMember(team.Id, userId); err != nil {
-		// user is not a member of team. Adding team member
-		if _, err := a.AddTeamMember(team.Id, userId); err != nil {
-			mlog.Error(fmt.Sprintf("Error occurred while adding user %s to team %s: %s", userId, team.Id, err.Error()))
-		}
+	team, err := a.GetTeamByName(teamName)
+	if err != nil {
+		return err
 	}
 
-	a.createAndJoinChannels(team.Id, lms.GetPublicChannelsToJoin(launchData), model.CHANNEL_OPEN, userId)
-	a.createAndJoinChannels(team.Id, lms.GetPrivateChannelsToJoin(launchData), model.CHANNEL_PRIVATE, userId)
+	publicChannels := a.createChannelsIfRequired(team.Id, lms.GetPublicChannelsToJoin(launchData), model.CHANNEL_OPEN)
+	privateChannels := a.createChannelsIfRequired(team.Id, lms.GetPrivateChannelsToJoin(launchData), model.CHANNEL_PRIVATE)
```

Comment:
Maybe we should not create a channel if no user is going to be added to them to avoid creating empty channels.

Reply:
Not relevant now as we are re-adding user memberships on sync.